### PR TITLE
knownfailuretest

### PR DESCRIPTION
**Pull request by thomasmesnard**
I'm not sure of what I did but I hope it will be ok.

Thanks again 

Best

Thomas

### CODE REVIEW

**Comment by thomasmesnard** [comment]
Travis Test is ok now.

Can you tell me if it's ok for you ?

Best,

**Comment by dwf** [comment]
@thomasmesnard This looks good to me. @nouiz is this what you had in mind? I think it's okay to merge.

File: theano/sandbox/gpuarray/tests/test_basic_ops.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def wrapper():
                 f()
             except Exception, e:
                 if isinstance(e, EClass):
-                    raise KnownFailureTest(msg, e)
+                    raise SkipTest("Not yet implemented")
```
Comment (nouiz) [inline theano/sandbox/gpuarray/tests/test_basic_ops.py]: Keep msg and e in the error.

File: theano/scan_module/tests/test_scan.py [hunk 2]
```diff
@@ -3173,9 +3173,9 @@ def lambda_fn(h, W1, W2):
                     if isinstance(x.op, theano.tensor.Elemwise)]) == 0
 
     def test_alloc_inputs2(self):
-        raise KnownFailureTest((
-            "This tests depends on an optimization for scan "
-            "that has not been implemented yet."))
+        raise SkipTest("Not yet implemented")
```
Comment (nouiz) [inline theano/scan_module/tests/test_scan.py]: You removed the old documentation at many places. Keep it everywhere except if you are sure the new one is better.

**Comment by nouiz** [comment]
2 small comments.

**Comment by lamblin** [comment]
Also, always skipping the test is probably not the answer.
Did you check each of them individually to see if the error was still occurring?

File: theano/tensor/tests/test_basic.py [hunk 3]
```diff
@@ -6026,9 +6026,8 @@ def test_arithmetic_cast(self):
                                     # may be a bug (investigation is currently
                                     # in progress), so in the meantime we just
                                     # mark this test as a known failure.
-                                    raise KnownFailureTest('Known issue with '
-                                            'numpy >= 1.6.x see #761')
-
+                                    raise SkipTest("Not yet implemented")
+                                    # Known issue with numpy >= 1.6.x see #761'
```
Comment (lamblin) [inline theano/tensor/tests/test_basic.py]: In particular, that one might already have been fixed.

File: theano/tests/test_rop.py [hunk 4]
```diff
@@ -21,6 +21,7 @@
 from theano.gof import Op, Apply
 from theano.gradient import grad_undefined
 from numpy.testing.noseclasses import KnownFailureTest
```
Comment (lamblin) [inline theano/tests/test_rop.py]: Remove the imports of `KnownFailureTest` if you are not using it anymore.

**Comment by lamblin** [comment]
There are also `KnownFailureTest` exceptions that are raised by the following decorator: `numpy.testing.dec.knownfailureif`, usually called with `@dec.knownfailureif`.
Those ones should be tested as well, and, if they are still failing, replaced by another decorator raising `SkipTest` instead.

File: theano/tensor/tests/test_basic.py (hunk 3 above)
Comment (abergeron) [inline theano/tensor/tests/test_basic.py]: As far as I know this still happens with 1.9.2.  It's something I've come against in my float16 work.

Comment (nouiz) [inline theano/tensor/tests/test_basic.py]: So it is probably by design. Thomas, can you update the comment to tell this?

**Comment by thomasmesnard** [comment]
Dear all,

Sorry for this delay. I had a very busy week. I have to finish a project first but I will try to fix these issues tomorrow.

Best

**Comment by thomasmesnard** [comment]
I think I fixed all the issues. I just would like to know if that seems good to you. 

I still have to fix the @dec.knownfailureif problem ..

Best

File: theano/compile/tests/test_function_module.py [hunk 5]
```diff
@@ -40,7 +40,8 @@ def test_none(self):
         fn = function([], None)  # ok
         rval = fn()
         if rval == []:
-            raise KnownFailureTest('See #254: Using None as function output leads to [] return value')
+            raise SkipTest("Not yet implemented")
+            # See #254: Using None as function output leads to [] return value
```
Comment (abergeron) [inline theano/compile/tests/test_function_module.py]: You should put the same string in the SkipTest message as there was in the KnownTestFailure message.

**Comment by abergeron** [comment]
I have one issue I noted, otherwise it's fine.  Also, this will need a rebase.

File: theano/compile/tests/test_function_module.py (hunk 5 above)
Comment (thomasmesnard) [inline theano/compile/tests/test_function_module.py]: Sorry, I missed this one. I'm fixing it today.

Thank's

**Comment by thomasmesnard** [comment]
Hi, 

here the new version with last modification and rebase.

Best

File: theano/tensor/tests/test_opt.py [hunk 6]
```diff
@@ -4284,8 +4282,8 @@ def test_local_log_erfc(self):
             0].op.scalar_op.fgraph.apply_nodes) == 22, len(f.maker.fgraph.toposort()[0].fgraph.toposort()[0].op.scalar_op.fgraph.apply_nodes)
         # TODO: fix this problem
         if theano.config.floatX == "float32" and theano.config.mode in ["DebugMode", "DEBUG_MODE"]:
-            raise KnownFailureTest(
-                "the python code upcast somewhere internally some value of float32 to python float for part of its computation. That make that the c and python code don't generate the same value. You can ignore this error.")
+            raise SkipTest('The python code upcast somewhere internally some value of float32 to python float for part of its computation.'
+                'That make that the c and python code dont generate the same value. You can ignore this error.')
```
Comment (lamblin) [inline theano/tensor/tests/test_opt.py]: `do not` or `don\'t` or use double quotes, but you should not skip apostrophes like that.

File: theano/tests/test_rop.py [hunk 7]
```diff
@@ -190,10 +190,8 @@ def check_rop_lop(self, y, out_shape):
         assert numpy.allclose(v1, v2), ('LOP mismatch: %s %s' % (v1, v2))
 
         if known_fail:
-            raise KnownFailureTest("Rop doesn't handle non-differentiable "
-                    "inputs correctly. Bug exposed by fixing Add.grad"
-                    " method.")
-
+            raise SkipTest('Rop doesn t handle non-differentiable inputs correctly.'
```
Comment (lamblin) [inline theano/tests/test_rop.py]: Same here.

File: theano/tensor/tests/test_opt.py [hunk 8]
```diff
@@ -1558,8 +1558,7 @@ def test_log_add():
         f([10000], [10000])  # causes overflow if handled incorrectly
         assert numpy.allclose(f([10000], [10000]), 20000)
     except AssertionError:
-        raise KnownFailureTest(('log(add(exp)) is not stabilized when adding '
-                'more than 2 elements, see #623'))
+        raise SkipTest("log(add(exp)) is not stabilized when adding more than 2 elements, see #623")
```
Comment (lamblin) [inline theano/tensor/tests/test_opt.py]: The reason this error message was on two lines was to avoid lines longer than 79 characters, as per pep8.

Comment (lamblin) [inline theano/tensor/tests/test_opt.py]: (That was not the only case, btw.)

**Comment by lamblin** [comment]
Also the `@dec.knownfailureif` are still not fixed.

**Comment by carriepl** [comment]
@thomasmesnard Can you take care of these few modifications so we can merge this PR and close the issue?

**Comment by carriepl** [comment]
This includes the changes requested by 5 different reviewers, Travis passes I and think this is fine as it is so I'm merging. Thanks for the PR.

**Pull request merged**